### ISSUE TITLE
Can anybody explain how the ncc loss function works?

### ISSUE THREAD

**Issue by tianmaliuixing** [voxelmorph/voxelmorph#135]
I see in the code: 
# compute local sums via convolution
        I_sum = conv_fn(I, sum_filt, strides, padding)
        J_sum = conv_fn(J, sum_filt, strides, padding)
        I2_sum = conv_fn(I2, sum_filt, strides, padding)
        J2_sum = conv_fn(J2, sum_filt, strides, padding)
        IJ_sum = conv_fn(IJ, sum_filt, strides, padding)

        # compute cross correlation
        win_size = np.prod(self.win)
        u_I = I_sum/win_size
        u_J = J_sum/win_size

        cross = IJ_sum - u_J*I_sum - u_I*J_sum + u_I*u_J*win_size
        I_var = I2_sum - 2 * u_I * I_sum + u_I*u_I*win_size
        J_var = J2_sum - 2 * u_J * J_sum + u_J*u_J*win_size

        cc = cross*cross / (I_var*J_var + self.eps)

        # return negative cc.
        return tf.reduce_mean(cc)

And in the article:
![image](https://user-images.githubusercontent.com/11666931/70618030-cebff400-1c4c-11ea-9957-2c81b67aecc2.png)

But I always feel that the code doesn't match the formula，can anybody explain the ncc loss function?

**Comment by adalca** [comment]
@tianmaliuixing , sorry for missing this earlier.

Can you explain which parts don't make sense? We use convolutions to obtain local sums, which are in the paper equation. If you expand the square, you'll need these sums for the fixed and moving image and their squares (f,m,f^2,m^2,f*m).
